matminer/featurizers/composition/roost.py

```diff
@@ -0,0 +1,569 @@
+import functools
+import os
+import numpy as np
+from matminer.featurizers.base import BaseFeaturizer
+from matminer.utils.data import MatscholarElementData
+
+# NOTE comprhys: is this really the only thing we can do here?
+# won't think not throw an error for someone without Roost installed
+# trying to use the RoostFeaturizer?
+try:
+    import torch
+    from torch.utils.data import Dataset, DataLoader
+    from torch.nn import L1Loss, MSELoss, CrossEntropyLoss
+    from torch.utils.tensorboard import SummaryWriter
+
+    from roost.core import Normalizer
+    from roost.roost.model import Roost
+
+    default_device = (
+        torch.device("cuda") if torch.cuda.is_available() else torch.device("cpu")
+    )
+except ImportError:
+    torch = None
+    Dataset, DataLoader = object, object
+    L1Loss, MSELoss, CrossEntropyLoss = object, object, object
+    SummaryWriter = object
+
+    Normalizer = object
+    Roost = object
+    default_device = None
+
+
+class RoostFeaturizer(BaseFeaturizer):
+    """
+    Representation learning from stiochiometry
+    """
+
+    def __init__(
+        self,
+        task="regression",
+        loss="L2",
+        model_name="roost",
+        elem_emb="matscholar",
+        elem_fea_len=64,
+        n_graph=3,
+        elem_heads=3,
+        elem_gate=[256],
+        elem_msg=[256],
+        cry_heads=3,
+        cry_gate=[256],
+        cry_msg=[256],
+        out_hidden=[1024, 512, 256, 128, 64],
+        n_targets=None,
+        run_id=1,
+        epochs=100,
+        optim="AdamW",
+        learning_rate=3e-4,
+        momentum=0.9,
+        weight_decay=1e-6,
+        batch_size=128,
+        workers=0,  # load data in main process -- allows caching
+        device=default_device,
+        **kwargs,
+    ):
+
+        # kill the featurizer if library not installed
+        if Roost == object:
+            raise ImportError(
+                "Roost library unavailable - please install and try again"
+            )
+
+        if task not in ["regression", "classification"]:
+            raise ValueError(
+                "Only 'regression' or 'classification' allowed for 'task'"
+            )
+
+        if n_targets is None:
+            if task == "regression":
+                n_targets = 1  # hard coded for single target regression
+            elif task == "classification":
+                n_targets = 2  # hard coded for binary clf
+
+        if elem_emb == "matscholar":
+            elem_emb_len = 200  # hard coded for matscholar
```

Comment:
So there are two things `elem_fea_len` and `elem_emb_len`, `elem_fea_len` is a hyperparameter for which 64 was the default I settled on and  `elem_emb_len` is the dimension of the initial representations used for the elements. The `matscholar` vectors are 200 long so I just hard coded this as I wanted to make use of the data store already in `matminer`. It would also be possible to add a `MatscholarElementData.len` property or something and set it using that. 

For big datasets the choice of initial embedding doesn't matter but for small datasets I saw `matscholar` typically performed slightly better than other things which is why I would be happy just to fix the choice of `matscholar` in the `matminer` stuff. If having used it here and it worked for them and they want to dig deeper then they can play around with the full code where they would have more freedom to set it up the model or add in tweaks etc.